### PR TITLE
Patch kdelibs for CVE-2019-14744

### PR DESCRIPTION
**Pull request by ttuegel**
###### Motivation for this change

See also: #70102 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Review by delroth — APPROVED** [review]

**Comment by gnidorah** [comment]
Please don't. I still use and maintain it. You may mark it as broken instead.

**Review by ghost — CHANGES_REQUESTED** [review]
Please don't. I still use and maintain it. You may mark it as broken instead.

**Comment by FRidh** [comment]
maintained yet broken? We can mark it as insecure, but really, it's a very old package. I think at this point you better use it from an older Nixpkgs version.

**Comment by gnidorah** [comment]
@FRidh Since there are no other known users of it... Actually @ttuegel made a patch for that CVE in #70102 that went into 19.03, why not merge it to master too? I would prefer to keep it since pykde5 is an old wip.

**Comment by ttuegel** [comment]
> Actually @ttuegel made a patch for that CVE in #70102 that went into 19.03, why not merge it to master too?

@gnidorah I didn't make the patch, it comes from upstream. I cherry-picked the changes from `release-19.03`, marked you as maintainer of both packages, and disabled the Hydra build.

File: pkgs/development/python-modules/pykde4/kdelibs.nix [hunk 1]
```diff
@@ -12,6 +12,14 @@ stdenv.mkDerivation rec {
     url = "mirror://kde/stable/applications/17.08.3/src/${pname}-${version}.tar.xz";
     sha256 = "1zn3yb09sd22bm54is0rn98amj0398zybl550dp406419sil7z9p";
   };
+  patches = [
+    # https://phabricator.kde.org/D22989
+    (fetchpatch {
+      url = "https://cgit.kde.org/kdelibs.git/patch/?id=2c3762feddf7e66cf6b64d9058f625a715694a00";
+      sha256 = "1wbzywh8lcc66n6y3pxs18h7cwkq6g216faz27san33jpl8ra1i9";
+      name = "kdelibs-D22989.patch";
```
Comment (delroth) [inline pkgs/development/python-modules/pykde4/kdelibs.nix]: CVE-2019-14744.patch (see https://github.com/NixOS/nixpkgs/issues/70102#issuecomment-537679033)

**Review by ghost — APPROVED** [review]
Thanks!

**Pull request merged**